solr/core/src/java/org/apache/solr/security/RuleBasedAuthorizationPlugin.java

```diff
@@ -16,329 +16,45 @@
  */
 package org.apache.solr.security;
 
-import java.io.IOException;
 import java.lang.invoke.MethodHandles;
 import java.security.Principal;
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.Function;
 
-import org.apache.solr.common.SpecProvider;
-import org.apache.solr.common.util.Utils;
-import org.apache.solr.common.util.ValidatingJsonMap;
-import org.apache.solr.common.util.CommandOperation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.unmodifiableMap;
-import static java.util.function.Function.identity;
-import static java.util.stream.Collectors.toMap;
-import static org.apache.solr.handler.admin.SecurityConfHandler.getListValue;
 import static org.apache.solr.handler.admin.SecurityConfHandler.getMapValue;
 
-
-public class RuleBasedAuthorizationPlugin implements AuthorizationPlugin, ConfigEditablePlugin, SpecProvider {
+/**
+ * Original implementation of Rule Based Authz plugin which configures user/role
+ * mapping in the security.json configuration
+ */
+public class RuleBasedAuthorizationPlugin extends RuleBasedAuthorizationPluginBase {
```

Comment:
Should this plugin have a new name, e.g. `ExplicitRolesRuleBasedAuthorizationPlugin`? So far I just kept the name to stay 100% back compat.

Reply:
Yea, planning for back compat is fine, I think.